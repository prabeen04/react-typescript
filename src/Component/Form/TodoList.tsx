import * as React from 'react'
import { ITodo } from './Form'
import TodoItem from "./TodoItem";
export interface ITodoListProps {
    todos: ITodo[];
    deleteTodo: (todo: ITodo) => void
}
export default function TodoList(props: ITodoListProps): JSX.Element {
    const { todos } = props
    return (
        <div>
            {
                !!todos && todos.map((todo: ITodo) => <TodoItem
                    key={todo.id}
                    todo={todo}
                    deleteTodo={props.deleteTodo} />)
            }
        </div>
    )
}
