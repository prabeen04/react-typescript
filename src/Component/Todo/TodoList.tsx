import * as React from 'react'
import { ITodo } from './Todo'
import TodoItem from "./TodoItem";
export interface ITodoListProps {
    todos: ITodo[];
    deleteTodo: (todo: ITodo) => void;
    toggleTodo: (todo: ITodo) => void;
    updateTodo: (todo: ITodo, newTodo: string) => void;
}
export default function TodoList(props: ITodoListProps): JSX.Element {
    const { todos } = props
    return (
        <ul className='list-group list-group-flush' data-test='TodoList'>
            {
                !!todos && todos.map((todo: ITodo) => <TodoItem
                    key={todo.id}
                    todo={todo}
                    deleteTodo={props.deleteTodo}
                    toggleTodo={props.toggleTodo}
                    updateTodo={props.updateTodo}
                />)
            }
        </ul>
    )
}
