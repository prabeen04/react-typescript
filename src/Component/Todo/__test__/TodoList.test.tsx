import * as React from 'react';
import { shallow } from "enzyme";
import TodoList from "../TodoList";


describe('<TodoList />', () => {
    let wrapper;
    let props = {
        todos: [],
        deleteTodo: jest.fn,
        updateTodo: jest.fn,
        toggleTodo: jest.fn,
    }
    beforeEach(() => {
        wrapper = shallow(<TodoList {...props} />)
    })
})