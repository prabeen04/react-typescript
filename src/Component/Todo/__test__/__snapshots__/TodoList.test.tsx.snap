// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    deleteTodo={[Function]}
    todos={Array []}
    toggleTodo={[Function]}
    updateTodo={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "todo-list-group",
      "data-test": "TodoList",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "todo-list-group",
        "data-test": "TodoList",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
